SmartDeviceLink-iOS/SmartDeviceLink/SDLIAPTransport.m

```diff
@@ -185,10 +185,9 @@ - (void)createIAPControlSessionWithAccessory:(EAAccessory *)accessory {
 
     if (self.controlSession) {
         self.controlSession.delegate = self;
-
-        if (self.protocolIndexTimer == nil) {
-            self.protocolIndexTimer = [[SDLTimer alloc] initWithDuration:protocolIndexTimeoutSeconds];
-        }
+        
+        // This will restart the timer if it already exists
+        self.protocolIndexTimer = [[SDLTimer alloc] initWithDuration:protocolIndexTimeoutSeconds];
 
```

Comment:
I'm pretty sure that this is a released object, not autoreleased. This should be a case where ARC puts in manual retain / release calls. An autorelease from a pool draining would just decrement the retain count by 1 anyway, the object's dealloc is enqueued to the runloop / queue as soon as the retain count reaches 0. Calling cancel would enqueue that call the same way.

That said, we may want to make the code more semantic / explicit that we're cancelling the timer by messaging 'cancel', but I'm very nearly positive that calling it will be no different than not, in this case.